client/packages/system/src/Item/Components/StockItemSearchInput/StockItemSearchInput.tsx

```diff
@@ -73,6 +73,10 @@ export const StockItemSearchInput: FC<StockItemSearchInputProps> = ({
     [selectedCode]
   );
 
+  useEffect(() => {
+    if (currentItem && search === '') setSearch(getOptionLabel(currentItem));
+  }, [currentItem]);
```

Comment:
Now that we manage independent search value (to allow for backspacing etc) 

search value was getting updated when you selected an item from the dropdown, but not when a currentItemId was passed in!